### ISSUE TITLE
stuck on login after enter auth code

### ISSUE THREAD

**Issue by cuimingda** [desktop/desktop#5533]
## Description
stuck on login after enter auth code in mac client.

## Version
* GitHub Desktop: 1.3.4
* Operating system: 10.13.6

## Steps to Reproduce
1. enter preference
2. sign in github
3. enter username and password
4. enter auth code
5. stuck on ....

### Expected Behavior
login succeed

### Actual Behavior
loading failed

## Additional Information
* I set hosts for github.com before, and clear just now. and have run `sudo killall -HUP mDNSResponder`
* i set 2fa today, just before login
* I have deleted github item in the keychain access.app

### Logs

there is errors in console
````
[AppStore] loading 0 repositories from store
app.tsx:210 App ready time: 155.05566599965096ms
app.tsx:211 Load time: 455.15861199982464ms
app.tsx:212 Renderer ready time: 142.58499999999998ms
install.ts:31 launching: 1.3.4 (Mac OS 10.13.6)
install.ts:31 execPath: '/Applications/GitHub Desktop.app/Contents/Frameworks/GitHub Desktop Helper.app/Contents/MacOS/GitHub Desktop Helper'
http.ts:135 POST https://api.github.com/authorizations 401 (Unauthorized)
t.request @ http.ts:135
i @ api.ts:621
async function (async)
i @ api.ts:612
authenticateWithBasicAuth @ sign-in-store.ts:279
_setSignInCredentials @ app-store.ts:3244
setSignInCredentials @ dispatcher.ts:717
onSubmit @ sign-in.tsx:74
onSubmit @ dialog.tsx:330
r @ react-dom.production.min.js:15
invokeGuardedCallback @ react-dom.production.min.js:16
invokeGuardedCallbackAndCatchFirstError @ react-dom.production.min.js:16
p @ react-dom.production.min.js:20
h @ react-dom.production.min.js:22
g @ react-dom.production.min.js:22
c @ react-dom.production.min.js:21
S @ react-dom.production.min.js:24
v @ react-dom.production.min.js:24
jt @ react-dom.production.min.js:83
batchedUpdates @ react-dom.production.min.js:187
nt @ react-dom.production.min.js:42
$t @ react-dom.production.min.js:85
interactiveUpdates @ react-dom.production.min.js:188
Xt @ react-dom.production.min.js:84
http.ts:135 GET https://github.com/user 410 (Gone)
t.request @ http.ts:135
request @ api.ts:495
fetchAccount @ api.ts:286
t.fetchUser @ api.ts:715
setTwoFactorOTP @ sign-in-store.ts:539
async function (async)
setTwoFactorOTP @ sign-in-store.ts:502
_setSignInOTP @ app-store.ts:3252
setSignInOTP @ dispatcher.ts:748
onSubmit @ sign-in.tsx:81
onSubmit @ dialog.tsx:330
r @ react-dom.production.min.js:15
invokeGuardedCallback @ react-dom.production.min.js:16
invokeGuardedCallbackAndCatchFirstError @ react-dom.production.min.js:16
p @ react-dom.production.min.js:20
h @ react-dom.production.min.js:22
g @ react-dom.production.min.js:22
c @ react-dom.production.min.js:21
S @ react-dom.production.min.js:24
v @ react-dom.production.min.js:24
jt @ react-dom.production.min.js:83
batchedUpdates @ react-dom.production.min.js:187
nt @ react-dom.production.min.js:42
$t @ react-dom.production.min.js:85
interactiveUpdates @ react-dom.production.min.js:188
Xt @ react-dom.production.min.js:84
install.ts:27 fetchAccount: failed with endpoint https://api.github.com
Error: Hello there, wayfaring stranger. If you’re reading this then you probably didn’t see our blog post a couple of years back announcing that this API would go away: http://git.io/17AROg Fear not, you should be able to get what you need from the shiny new Events API instead.
    at Object.t.parsedResponse (/webpack:/app/src/lib/http.ts:161:4)
    at <anonymous>
warn @ install.ts:27
fetchAccount @ api.ts:290
async function (async)
fetchAccount @ api.ts:284
t.fetchUser @ api.ts:715
setTwoFactorOTP @ sign-in-store.ts:539
async function (async)
setTwoFactorOTP @ sign-in-store.ts:502
_setSignInOTP @ app-store.ts:3252
setSignInOTP @ dispatcher.ts:748
onSubmit @ sign-in.tsx:81
onSubmit @ dialog.tsx:330
r @ react-dom.production.min.js:15
invokeGuardedCallback @ react-dom.production.min.js:16
invokeGuardedCallbackAndCatchFirstError @ react-dom.production.min.js:16
p @ react-dom.production.min.js:20
h @ react-dom.production.min.js:22
g @ react-dom.production.min.js:22
c @ react-dom.production.min.js:21
S @ react-dom.production.min.js:24
v @ react-dom.production.min.js:24
jt @ react-dom.production.min.js:83
batchedUpdates @ react-dom.production.min.js:187
nt @ react-dom.production.min.js:42
$t @ react-dom.production.min.js:85
interactiveUpdates @ react-dom.production.min.js:188
Xt @ react-dom.production.min.js:84
install.ts:27 fetchUser: failed with endpoint https://api.github.com
Error: Hello there, wayfaring stranger. If you’re reading this then you probably didn’t see our blog post a couple of years back announcing that this API would go away: http://git.io/17AROg Fear not, you should be able to get what you need from the shiny new Events API instead.
    at Object.t.parsedResponse (/webpack:/app/src/lib/http.ts:161:4)
    at <anonymous>
warn @ install.ts:27
t.fetchUser @ api.ts:732
async function (async)
t.fetchUser @ api.ts:710
setTwoFactorOTP @ sign-in-store.ts:539
async function (async)
setTwoFactorOTP @ sign-in-store.ts:502
_setSignInOTP @ app-store.ts:3252
setSignInOTP @ dispatcher.ts:748
onSubmit @ sign-in.tsx:81
onSubmit @ dialog.tsx:330
r @ react-dom.production.min.js:15
invokeGuardedCallback @ react-dom.production.min.js:16
invokeGuardedCallbackAndCatchFirstError @ react-dom.production.min.js:16
p @ react-dom.production.min.js:20
h @ react-dom.production.min.js:22
g @ react-dom.production.min.js:22
c @ react-dom.production.min.js:21
S @ react-dom.production.min.js:24
v @ react-dom.production.min.js:24
jt @ react-dom.production.min.js:83
batchedUpdates @ react-dom.production.min.js:187
nt @ react-dom.production.min.js:42
$t @ react-dom.production.min.js:85
interactiveUpdates @ react-dom.production.min.js:188
Xt @ react-dom.production.min.js:84
http.ts:50 Uncaught (in promise) s: Hello there, wayfaring stranger. If you’re reading this then you probably didn’t see our blog post a couple of years back announcing that this API would go away: http://git.io/17AROg Fear not, you should be able to get what you need from the shiny new Events API instead.
    at Object.t.parsedResponse (file:///Applications/GitHub%20Desktop.app/Contents/Resources/app/renderer.js:27:38143)
    at <anonymous>
````

**Comment by Anondobangla** [comment]
how to solve

**Comment by shiftkey** [comment]
@cuimingda it looks like your somehow able to trick the app to call a different GitHub API host, and the URL in here goes to [this blog post](https://blog.github.com/2012-06-12-github-api-v2-end-of-life/) from 2012 about how our v1 and v2 APIs are deprecated:

```
install.ts:27 fetchAccount: failed with endpoint https://api.github.com
Error: Hello there, wayfaring stranger. If you’re reading this then you probably didn’t see our blog post a couple of years back announcing that this API would go away: http://git.io/17AROg Fear not, you should be able to get what you need from the shiny new Events API instead.
    at Object.t.parsedResponse (/webpack:/app/src/lib/http.ts:161:4)
    at <anonymous>
warn @ install.ts:27
```

> I set hosts for github.com before, and clear just now.

Why do you need to set these?

**Comment by cuimingda** [comment]
@shiftkey I set hosts for github.com before, the ip is fetched from http://github.com.ipaddress.com/ , but now I have already removed these settings in /etc/hosts. China has blocked some IPs of github, so I set some accessable IP address.

after removed settings in /etc/hosts, I do not make other custom settings. The install dmg is download from https://desktop.github.com/

I set ssh key in github.com, and I can use github service from shell normally. I don't know why the offical client would use a api deprecated in 2012.

**Comment by cuimingda** [comment]
and I can use GitHub Desktop to commit normally.
but push and fetch is forbidden, because require log in .

**Comment by cuimingda** [comment]
https://ibb.co/dcaWaK
here is my login error capture

**Comment by shiftkey** [comment]
>  I don't know why the offical client would use a api deprecated in 2012.

@cuimingda Desktop uses `api.github.com` for fetching the data it needs, not `github.com`. I'm not sure if you're setting that too, so that might be why it's behaving differently.

What does `nslookup api.github.com` resolve to for you? Here's my machine's output:

```
$ nslookup api.github.com
Server:		127.0.0.1
Address:	127.0.0.1#53

Non-authoritative answer:
Name:	api.github.com
Address: 192.30.253.116
Name:	api.github.com
Address: 192.30.253.117
```

**Comment by cuimingda** [comment]
here is mine, 223.5.5.5 is Aliyun DNS, and the 3 IPs all are github's?
````
> nslookup api.github.com
Server:		223.5.5.5
Address:	223.5.5.5#53

Non-authoritative answer:
Name:	api.github.com
Address: 13.250.168.23
Name:	api.github.com
Address: 13.250.94.254
Name:	api.github.com
Address: 54.169.195.247
````

**Comment by cuimingda** [comment]
@shiftkey thks for your help, just saying I just solved this problem.

I removed some config in ~/etc/.ssh/config
````
  AddKeysToAgent yes
  IdentityFile ~/.ssh/id_rsa
````

I also removed some content in ~/etc/.ssh/known_hosts, which contains github.com

I uninstalled GitHub Desktop by CleanMyMac, including all user config files, and reinstall from desktop.github.com

now it is OK, I can login github and push. But I still don't know which is the key operation helping me to fix the problem.
